packages/transaction-controller/src/utils/transaction-type.ts

```diff
@@ -42,16 +41,19 @@ export async function determineTransactionType(
 
   const hasValue = txParams.value && Number(txParams.value) !== 0;
 
-  const tokenMethodName = [
-    TransactionType.tokenMethodApprove,
-    TransactionType.tokenMethodSetApprovalForAll,
-    TransactionType.tokenMethodTransfer,
-    TransactionType.tokenMethodTransferFrom,
-    TransactionType.tokenMethodSafeTransferFrom,
-  ].find((methodName) => methodName.toLowerCase() === name?.toLowerCase());
-
-  if (data && tokenMethodName && !hasValue) {
-    return { type: tokenMethodName, getCodeResponse: resultCode };
+  const name = parseStandardTokenTransactionData(data)?.name;
+  if (name !== undefined) {
+    const tokenMethodName = [
+      TransactionType.tokenMethodApprove,
+      TransactionType.tokenMethodSetApprovalForAll,
+      TransactionType.tokenMethodTransfer,
+      TransactionType.tokenMethodTransferFrom,
+      TransactionType.tokenMethodSafeTransferFrom,
+    ].find((methodName) => methodName.toLowerCase() === name.toLowerCase());
+
+    if (data && tokenMethodName && !hasValue) {
```

Comment:
Rather than nesting conditions, should we use early returns for better readability?

Also based on the `data` and `hasValue`, could we have also returned before even getting the name?

Something like:

```
  const { data, to } = txParams;

  if (data && !to) {
    return { type: TransactionType.deployContract, getCodeResponse: undefined };
  }

  const { contractCode: getCodeResponse, isContractAddress } =
    await readAddressAsContract(ethQuery, to);

  if (!isContractAddress) {
    return { type: TransactionType.simpleSend, getCodeResponse };
  }

  const hasValue = txParams.value && Number(txParams.value) !== 0;

  const contractInteractionResult = {
    type: TransactionType.contractInteraction,
    getCodeResponse,
  };

  if (!data || hasValue) {
    return contractInteractionResult;
  }

  const name = parseStandardTokenTransactionData(data)?.name;

  if (!name) {
    return contractInteractionResult;
  }

  const tokenMethodName = [
    TransactionType.tokenMethodApprove,
    TransactionType.tokenMethodSetApprovalForAll,
    TransactionType.tokenMethodTransfer,
    TransactionType.tokenMethodTransferFrom,
    TransactionType.tokenMethodSafeTransferFrom,
  ].find((methodName) => methodName.toLowerCase() === name.toLowerCase());

  if (tokenMethodName) {
    return { type: tokenMethodName, getCodeResponse };
  }

  return contractInteractionResult;
```